nixos/modules/services/audio/navidrome.nix

```diff
@@ -1,84 +1,140 @@
-{ config, lib, pkgs, ... }:
-
-with lib;
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
 
 let
   cfg = config.services.navidrome;
-  settingsFormat = pkgs.formats.json {};
-in {
+  settingsFormat = pkgs.formats.json { };
+in
+{
   options = {
-    services.navidrome = {
+    services.navidrome =
+      let
+        inherit (lib) mkEnableOption mkPackageOption mkOption;
+        inherit (lib.types) bool str;
+      in
+      {
 
-      enable = mkEnableOption (lib.mdDoc "Navidrome music server");
+        enable = mkEnableOption "Navidrome music server";
 
-      package = mkPackageOption pkgs "navidrome" { };
+        package = mkPackageOption pkgs "navidrome" { };
 
-      settings = mkOption rec {
-        type = settingsFormat.type;
-        apply = recursiveUpdate default;
-        default = {
-          Address = "127.0.0.1";
-          Port = 4533;
+        user = mkOption {
+          type = str;
+          default = "navidrome";
+          description = "User account under which Navidrome runs.";
         };
-        example = {
-          MusicFolder = "/mnt/music";
+
+        group = mkOption {
+          type = str;
+          default = "navidrome";
+          description = "Group under which Navidrome runs.";
+        };
+
+        settings = mkOption {
+          type = settingsFormat.type;
+          default = {
+            Address = "127.0.0.1";
+            Port = 4533;
+          };
+          example = {
+            MusicFolder = "/mnt/music";
+          };
+          description = "Configuration for Navidrome, see <https://www.navidrome.org/docs/usage/configuration-options/> for supported values.";
         };
-        description = lib.mdDoc ''
-          Configuration for Navidrome, see <https://www.navidrome.org/docs/usage/configuration-options/> for supported values.
-        '';
-      };
 
-      openFirewall = mkOption {
-        type = types.bool;
-        default = false;
-        description = lib.mdDoc "Whether to open the TCP port in the firewall";
+        openFirewall = mkOption {
+          type = bool;
+          default = false;
+          description = "Whether to open the TCP port in the firewall";
+        };
       };
-    };
   };
 
-  config = mkIf cfg.enable {
-    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [cfg.settings.Port];
+  config =
+    let
+      inherit (lib) mkIf getExe;
+      WorkingDirectory = "/var/lib/navidrome";
+    in
+    mkIf cfg.enable {
+
+      systemd = {
+        tmpfiles.settings.navidromeDirs = {
+          "${cfg.settings.DataFolder or WorkingDirectory}"."d" = {
+            mode = "700";
+            inherit (cfg) user group;
+          };
+          "${cfg.settings.CacheFolder or WorkingDirectory + "/cache"}"."d" = {
```

Comment:
Either use a string interpolation or parenthesis to make this clearer.

Reply:
`(WorkingDirectory + "/cache")` sound alright?